config/routes.rb

```diff
@@ -57,61 +57,48 @@
   end
 
   namespace :admin do
-    resources :suppliers, only: %i[index new]
-    scope "suppliers/new" do
-      post "/", controller: :suppliers, action: :receive_new, as: :new_supplier
-      get "supplier-type", controller: :suppliers, action: :new_supplier_type, as: :new_supplier_type
-      post "supplier-type", controller: :suppliers, action: :receive_new_supplier_type
-
-      scope "lead-provider" do
-        get "choose-cip", controller: :lead_providers, action: :choose_cip, as: :new_lead_provider_cip
-        post "choose-cip", controller: :lead_providers, action: :receive_cip
-        get "choose-cohorts", controller: :lead_providers, action: :choose_cohorts, as: :new_lead_provider_cohorts
-        post "choose-cohorts", controller: :lead_providers, action: :receive_cohorts
-        get "review", controller: :lead_providers, action: :review, as: :new_lead_provider_review
-        post "/", controller: :lead_providers, action: :create, as: :create_lead_provider
-        get "success", controller: :lead_providers, action: :success, as: :new_lead_provider_success
+    scope :suppliers, module: "suppliers" do
```

Comment:
This I really like i.e. using resources routes as much as possible. Looks a lot cleaner and requires less customisation.